### PR TITLE
Add the content-type driver to PHPCR-ODM

### PR DESCRIPTION
**Pull request by dantleech**

### CODE REVIEW

File: src/DependencyInjection/Compiler/FormExtensionPass.php [hunk 1]
```diff
@@ -16,5 +17,6 @@ public function process(ContainerBuilder $container)
         $registryDef = $container->getDefinition('form.registry');
         $extensions = $registryDef->getArgument(0);
         $extensions[] = new Reference('cmf_content_type.form.extension.field');
+        $registryDef->replaceArgument(0, $extensions);
```
Comment (dantleech) [inline src/DependencyInjection/Compiler/FormExtensionPass.php:20]: Add test for this.

File: src/DependencyInjection/Compiler/DoctrinePhpcrOdmDriverPass.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+        if ($class !== MappingDriverChain::class) {
+            throw new \RuntimeException(sprintf(
+                'PHPCR-ODM is not using the chain metadata driver, got "%s"',
+                $driverDef->getClass()
+            ));
+        }
+
+        $driverDef->addMethodCall('addDriver', [
+            new Reference('cmf_content_type.doctrine.phpcr_odm.field_driver'),
```
Comment (dantleech) [inline src/DependencyInjection/Compiler/DoctrinePhpcrOdmDriverPass.php]: @dbu do you know if there is a better way to add a new metadata driver?

Comment (dbu) [reply src/DependencyInjection/Compiler/DoctrinePhpcrOdmDriverPass.php]: i don't know. did you check the phpcr-odm bundle for service tags or other things we might use? if not it looks not that bad, only relying on implicit things.

ah hm, default in the service name: does that mean we have one of those metadata driver services per object manager?

Comment (dantleech) [reply src/DependencyInjection/Compiler/DoctrinePhpcrOdmDriverPass.php]: yeah this should be done for each manager. looking at the main abstract DoctrineExtension looks like there might be a better way to do it.

File: src/DependencyInjection/Storage/Compiler/DoctrinePhpcrOdmDriverPass.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace Psi\Bundle\ContentType\DependencyInjection\Storage\Compiler;
+
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\Reference;
+use Doctrine\Common\Persistence\Mapping\Driver\MappingDriverChain;
+
+class DoctrinePhpcrOdmDriverPass implements CompilerPassInterface
```
Comment (dantleech) [inline src/DependencyInjection/Storage/Compiler/DoctrinePhpcrOdmDriverPass.php:10]: Add comment

**Pull request merged**